esphome/components/light/base_light_effects.h

```diff
@@ -57,14 +57,23 @@ class RandomLightEffect : public LightEffect {
     if (now - this->last_color_change_ < this->update_interval_) {
       return;
     }
+
+    auto color_mode = this->state_->remote_values.get_color_mode();
     auto call = this->state_->turn_on();
-    if (this->state_->get_traits().get_supports_rgb()) {
-      call.set_red_if_supported(random_float());
-      call.set_green_if_supported(random_float());
-      call.set_blue_if_supported(random_float());
-      call.set_white_if_supported(random_float());
+    if (*color_mode & *ColorCapability::RGB) {
+      call.set_red(random_float());
+      call.set_green(random_float());
+      call.set_blue(random_float());
+    } else if (*color_mode & *ColorCapability::COLOR_TEMPERATURE) {
```

Comment:
```suggestion
    }
    if (*color_mode & *ColorCapability::COLOR_TEMPERATURE) {
```


These should be ˋifˋs rather than else ifs

Reply:
I'm not sure that's desired here. For RGBW/RGBWW lights, it would result in the white channel always being on, so you won't get all colors, but only low-saturation colors. See e.g. esphome/issues#1830. 

It's probably a matter of preference though.